torrent.go

```diff
@@ -796,13 +796,16 @@ func (t *Torrent) close() (err error) {
 	t.closed.Set()
 	t.tickleReaders()
 	if t.storage != nil {
-		go func() {
+		func() {
```

Comment:
This is done asynchronously to avoid a situation where locks are obtained in the wrong order. Was that considered?

Reply:
No, sorry. I saw that it was added on July 17, and the commit just mentioned "Close torrent storage asynchronously on drop". The reason I was trying to avoiding the async here is because when using cmd/torrent, when download ends or Ctrl-C is input, execution ends before my storage.Close() has a chance to finish cleaning up. I figured moving the async further up the call stack (only place I see ".close()" being called) would be equivalent. Would it be better to keep the goroutine as-is but have (*Torrent) close() wait on it's conclusion before returning?
```suggestion
	var errorReturn chan error
	if t.storage != nil {
	errorReturn = make(chan error, 1)
		go func() {
			t.storageLock.Lock()
			defer t.storageLock.Unlock()
			if f := t.storage.Close; f != nil {
				errorReturn <- f()
			}
		}()
	}
	...
	if errorReturn != nil {
	     err = <- errorReturn 
	}
	return ```

Reply:
No, I think the issue is that the storage close can deadlock while Torrent.close holds any locks. What if in the Client, where it drops all the torrents, it signals to drop everything, marks itself closed, then releases the client lock and waits for the drops to complete. It might be easiest to pass a waitgroup or error group into the drop/close methods so the caller has the final say about when to wait. In the current circumstance, I think the storage close goroutine would signal when it's done.